assets/ui/marketScreen.ui

```diff
@@ -0,0 +1,176 @@
+                  "use-content-height": false,
+                  "position-left": {
+                    "target": "LEFT"
+                  },
+                  "position-right": {
+                    "target": "RIGHT"
+                  }
+                },
+                "contents": [
+                  {
+                    "type": "FlowLayout",
+                    "id": "marketItemList",
+                    "layoutInfo": {
+                      "use-content-height": false,
+                      "use-content-width": false
+                    }
+                  }
+                ]
+              }
+            ]
+          },
+          {
+            "type": "RelativeLayout",
+            "contents": [
+              {
+                "type": "UILabel",
+                "id": "itemName",
+                "layoutInfo": {
+                  "use-content-height": true,
+                  "use-content-width": true,
+                  "position-top": {
+                    "offset": 16,
+                    "target": "TOP",
+                    "widget": "columnsHolder"
+                  },
+                  "position-horizontal-center": {}
+                }
+              },
+              {
+                "type": "ItemIcon",
+                "id": "itemDisplayIcon",
+                "layoutInfo": {
+                  "height": 50,
+                  "weight": 50,
+                  "use-content-width" : false,
+                  "position-horizontal-center": {},
+                  "position-top": {
+                    "offset": 16,
+                    "target": "BOTTOM",
+                    "widget": "itemName"
+                  }
+                }
+              },
+              {
+                "type": "ScrollableArea",
+                "id": "itemInfoPlaceholder",
+                "layoutInfo": {
+                  "position-top": {
+                    "target": "BOTTOM",
+                    "widget": "itemDisplayIcon",
+                    "offset": 16
+                  },
+                  "height": 200
+                },
+                "contents": [
+                  {
+                    "type": "UILabel",
+                    "id": "itemDescription",
+                    "layoutInfo": {
+                      "use-content-height": true,
+                      "use-content-width": true
+                    }
+                  }
+                ]
+              },
+              {
+                "type": "RelativeLayout",
+                "id": "buttonPlaceholder",
+                "layoutInfo": {
+                  "position-top": {
+                    "widget": "itemInfoPlaceholder",
+                    "target": "BOTTOM"
+                  },
+                  "position-bottom": {
+                    "target": "BOTTOM"
+                  }
+                },
+                "contents": [
+                  {
+                    "type": "UIButton",
+                    "id": "buyButton",
```

Comment:
seems not to be hooked anywhere. buy doesn't work :(

Reply:
saw your comment on slack that you implemented the buying on a separate branch.
would be great if you could include that info in the PR description the next time to avoid expecting something different than what the PR actually brings ;)